MarkBind/markbind
# Support Page Navigation Sidebar

**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [x] Documentation update
• [x] New feature


Resolves #219 



**What is the rationale for this request?**
User would like to be able to use a page navigation component that is automatically generated from headings within the page.

Prototype 2 Demo |
--- |
![page-nav-prototype-2](https://user-images.githubusercontent.com/31084833/49989973-f8e9c200-ffb6-11e8-9dc9-4c4c0d23eb11.gif) |

Prototype 3, Responsive Design Demo |
--- |
![page-nav-responsive](https://user-images.githubusercontent.com/31084833/50473448-e67c7900-09f7-11e9-8734-527cbf5c08ff.gif)



To Do | Done?
--- | ---
Responsive design | ✔️  
Change specifying `pageNav` within `front-matter` from `Boolean` to `{1 ... 6} or default` | ✔️ 
Add specifying `pageNavHeader` to `front-matter` as the header text for page navigation | ✔️ 
Adaptive CSS if `site-nav` or `footer` is not specified in the page | ✔️  
Page navigation automatically scrolls to active heading as user scroll the page | ✔️ 
Indentation of smaller headings | ✔️  
Include headings in `Panel headers` and `Panel contents` if expanded  | ✔️ 
Live preview via `markbind serve` re-inserts page navigation | ✔️ 
Update test site to use page navigation | ✔️ 
Change page navigation and scroll bar styling from default Bootstrap theme | ✔️ 
Update documentation for new feature and using it | ✔️ 
Support auto expanding and collapsing of panels upon clicking an anchor link | ⚠️<br> On hold 

 
References |
--- |
[Bootstrap Scrollspy](https://getbootstrap.com/docs/4.0/components/scrollspy/) |
[Custom Scrollbars](https://css-tricks.com/custom-scrollbars-in-webkit/) |
[Scrollspy Event Handling](https://stackoverflow.com/questions/48693913/bootstrap-4-activate-bs-scrollspy-event-is-not-firing) |

**Testing instructions:**
1. Use Netlify Preview
1. Go to the "Docs" tab then "Using Components" sub section and the page navigation bar should appear.
